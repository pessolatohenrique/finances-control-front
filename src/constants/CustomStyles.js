import { styled } from "@mui/material/styles";
import { tableCellClasses } from "@mui/material/TableCell";
import { TableCell } from "@mui/material";

export const StyledTableCell = styled(TableCell)(({ theme }) => ({
  [`&.${tableCellClasses.head}`]: {
    fontSize: "1.1em",
    fontWeight: 300,
  },
}));
