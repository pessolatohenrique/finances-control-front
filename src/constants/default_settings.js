export const PROJECT_NAME = "Project Name";
export const THEME_COLOR = "#1976d2";
export const SNACKBAR_DIRECTION = { vertical: "top", horizontal: "right" };
