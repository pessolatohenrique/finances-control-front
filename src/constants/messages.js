export const REQUIRED_MESSAGE = "Campo de preenchimento obrigat√≥rio";
export const USER_NOT_EXISTS_MESSAGE = "This user doesnt exists";
export const WRONG_PASSWORD_MESSAGE = "Invalid username or password";
