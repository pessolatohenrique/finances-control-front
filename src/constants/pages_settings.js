export const PATHS_WITHOUT_LOGIN = ["/usuario/novo"];
