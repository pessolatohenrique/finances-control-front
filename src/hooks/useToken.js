import { useState } from "react";
import { initializeAxios } from "../utils/requests";

function useToken() {
  const [token, setToken] = useState(getToken());
  const [refreshToken, setRefreshToken] = useState(getRefreshToken());

  function getToken() {
    const token = localStorage.getItem("token");
    return token || "";
  }

  function getRefreshToken() {
    const refreshToken = localStorage.getItem("refreshToken");
    return refreshToken || "";
  }

  async function saveToken(token) {
    localStorage.setItem("token", token);
    initializeAxios(token);
    setToken(token);
  }

  async function saveRefreshToken(token) {
    localStorage.setItem("refreshToken", token);
    setRefreshToken(token);
  }

  return {
    token,
    refreshToken,
    setToken: saveToken,
    setRefreshToken: saveRefreshToken,
  };
}

export default useToken;
