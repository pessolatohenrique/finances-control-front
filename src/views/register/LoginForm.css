body {
  overflow: hidden;
}

#root {
  background-image: url("../../assets//moneyfinance-1.jpg");
}
