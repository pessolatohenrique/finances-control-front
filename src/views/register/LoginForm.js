import React from "react";
import {
  Card,
  CardContent,
  Typography,
  Grid,
  TextField,
  Button,
  Snackbar,
  Alert,
} from "@mui/material";
import { useForm } from "react-hook-form";
import axios from "axios";
import {
  THEME_COLOR,
  SNACKBAR_DIRECTION,
} from "../../constants/default_settings";
import { REQUIRED_MESSAGE } from "../../constants/messages";
import moneyImage from "../../assets/moneyfinance-1.jpg";
import useToast from "../../hooks/useToast";

function LoginForm({ setToken, setRefreshToken }) {
  const { open, error, setError, showToast, hideToast } = useToast();

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  async function loginUser(credentials) {
    try {
      const response = await axios.post(
        `${process.env.REACT_APP_API_URL}/login`,
        { ...credentials }
      );

      setToken(response.data.accessToken);
      setRefreshToken(response.data.refreshToken);
    } catch (error) {
      showToast();
      setError(error.response.data);
    }
  }

  const onSubmit = (data) => loginUser(data);

  return (
    <>
      {/* <Container> */}
      <Grid
        display="flex"
        flexDirection="column"
        flexGrow={5}
        justifyContent="center"
        alignItems="center"
        minHeight="100vh"
        sx={{
          backgroundImage: `url(${moneyImage})`,
          overflow: "hidden",
        }}
      >
        <Snackbar
          anchorOrigin={SNACKBAR_DIRECTION}
          open={open}
          autoHideDuration={6000}
          onClose={hideToast}
        >
          <Alert onClose={hideToast} severity="error" sx={{ width: "100%" }}>
            {error}
          </Alert>
        </Snackbar>
        <Card sx={{ width: "50%" }}>
          <CardContent>
            <Typography
              gutterBottom
              variant="h5"
              component="h1"
              color={THEME_COLOR}
            >
              Entrar
            </Typography>
            <Typography
              gutterBottom
              variant="p"
              component="p"
              //   color={THEME_COLOR}
            >
              Utilize as suas credenciais para acessar o sistema
            </Typography>
            <form onSubmit={handleSubmit(onSubmit)}>
              <Grid container direction={"column"} spacing={2}>
                <Grid item>
                  <TextField
                    fullWidth
                    id="username"
                    label="Usuário"
                    inputProps={{ "data-testid": "username" }}
                    error={Boolean(errors.username)}
                    helperText={errors.username && REQUIRED_MESSAGE}
                    {...register("username", {
                      required: true,
                    })}
                  />
                </Grid>
                <Grid item>
                  <TextField
                    fullWidth
                    id="password"
                    label="Senha"
                    inputProps={{
                      "data-testid": "password",
                      type: "password",
                    }}
                    error={Boolean(errors.password)}
                    helperText={errors.password && REQUIRED_MESSAGE}
                    {...register("password", {
                      required: true,
                    })}
                  />
                </Grid>

                <Grid item>
                  <Button
                    variant="contained"
                    fullWidth
                    type="submit"
                    data-testid="submit-button"
                  >
                    Entrar
                  </Button>

                  <Typography
                    gutterBottom
                    variant="p"
                    component="p"
                    //   color={THEME_COLOR}
                  >
                    <br />
                    Ainda não tem uma conta? Cadastre-se{" "}
                    <a href="/usuario/novo">aqui</a>
                  </Typography>
                </Grid>
              </Grid>
            </form>
          </CardContent>
        </Card>
      </Grid>
      {/* </Container> */}
    </>
  );
}
export default LoginForm;
