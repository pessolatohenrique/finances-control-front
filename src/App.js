import React from "react";
import AppBarWrapper from "./components/AppBarWrapper";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import ExampleList from "./views/register/ExampleList";
import ExampleForm from "./views/register/ExampleForm";
import ExampleDetail from "./views/register/ExampleDetail";
import NotFound from "./components/NotFound";
import LoginForm from "./views/register/LoginForm";

function App() {
  return (
    <Router>
      <AppBarWrapper />

      <Switch>
        <Route exact path="/signin">
          <LoginForm />
        </Route>

        <Route exact path="/livros">
          <ExampleList />
        </Route>

        <Route exact path="/autores/novo">
          <ExampleForm />
        </Route>

        <Route exact path="/livros/:id">
          <ExampleDetail />
        </Route>

        <Route path="*">
          <NotFound />
        </Route>
      </Switch>
    </Router>
  );
}

export default App;
