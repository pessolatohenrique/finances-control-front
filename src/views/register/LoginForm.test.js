import {
  render,
  screen,
  fireEvent,
  waitForElement,
  waitForDomChange,
  waitFor,
} from "@testing-library/react";
import { renderHook } from "@testing-library/react-hooks";
import { PROJECT_NAME } from "../../constants/default_settings";
import {
  USER_NOT_EXISTS_MESSAGE,
  WRONG_PASSWORD_MESSAGE,
} from "../../constants/messages";
import App from "../../App";
import LoginForm from "./LoginForm";
import useToken from "../../hooks/useToken";

describe("Login Form", () => {
  it("should generate snapshot", () => {
    const { setToken } = renderHook(() => useToken());
    const { container } = render(<LoginForm setToken={setToken} />);
    expect(container).toMatchSnapshot();
  });

  it("should login when using valid credentials", async () => {
    render(<App />);
    const submitBtn = screen.getByTestId("submit-button");
    const username = screen.getByTestId("username");
    const password = screen.getByTestId("password");

    fireEvent.input(username, {
      target: { value: process.env.REACT_APP_API_DEFAULT_USER },
    });
    fireEvent.input(password, {
      target: { value: process.env.REACT_APP_API_DEFAULT_PASSWORD },
    });
    fireEvent.click(submitBtn);

    await waitForElement(() => screen.getByText(PROJECT_NAME));
  });

  it("should not login when user not exists", async () => {
    const { setToken } = renderHook(() => useToken());
    render(<LoginForm setToken={setToken} />);

    const submitBtn = screen.getByTestId("submit-button");
    const username = screen.getByTestId("username");
    const password = screen.getByTestId("password");

    fireEvent.input(username, {
      target: { value: "usuarioquenaoexiste" },
    });
    fireEvent.input(password, {
      target: { value: process.env.REACT_APP_API_DEFAULT_PASSWORD },
    });
    fireEvent.click(submitBtn);

    await waitFor(() => expect(screen.getByText(USER_NOT_EXISTS_MESSAGE)));
  });

  it("should not login when password is invalid", async () => {
    const { setToken } = renderHook(() => useToken());
    render(<LoginForm setToken={setToken} />);

    const submitBtn = screen.getByTestId("submit-button");
    const username = screen.getByTestId("username");
    const password = screen.getByTestId("password");

    fireEvent.input(username, {
      target: { value: process.env.REACT_APP_API_DEFAULT_USER },
    });
    fireEvent.input(password, {
      target: { value: "incorrectpassword" },
    });
    fireEvent.click(submitBtn);

    await waitFor(() => expect(screen.getByText(WRONG_PASSWORD_MESSAGE)));
  });
});
