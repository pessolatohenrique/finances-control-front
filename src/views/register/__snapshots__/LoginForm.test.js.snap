// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form should generate snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root css-136s2ky-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-b2pbdi-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
        >
          Entrar
        </h1>
        <p
          class="MuiTypography-root MuiTypography-p MuiTypography-gutterBottom css-anmq6p-MuiTypography-root"
        >
          Utilize as suas credenciais para acessar o sistema
        </p>
        <form>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1rl1og4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="username"
                  id="username-label"
                >
                  Usuário
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="username"
                    id="username"
                    name="username"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Usuário
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="password"
                  id="password-label"
                >
                  Senha
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="password"
                    id="password"
                    name="password"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Senha
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
                data-testid="submit-button"
                tabindex="0"
                type="submit"
              >
                Entrar
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
